src/client/base-client.ts

```diff
@@ -35,8 +37,25 @@ export abstract class BaseClient<T extends {}> {
     this.client = createClient<T>({ baseUrl: this.baseUrl });
     const authMiddleware: Middleware = {
       onRequest: async ({ request }) => {
-        if (!this.token && this.getTokenFn) {
+        if (
+          (!this.token || Date.now() >= this.tokenExpiryCheckTime) &&
```

Comment:
I would do
```suggestion
          (!this.token || ((this.tokenExpiryCheckTime - Date.now()) <= 5000) &&
```
In this way we can take a 5 seconds range in which we can prevent the token to expire while we are using it

Reply:
Rather do `Date.now() >= (this.tokenExpiryCheckTime - expiryThresholdLimiterMs` or something?

Reply:
should be `+`, not `-`
Anyway, as you prefer

Reply:
If we want to be sure the token is not expired, don't we want to remove a few secs from expiry time, to make sure we get a new one before the current expires?

Reply:
the result would be the same, so as you prefer